app/mm-redux/selectors/entities/groups.ts

```diff
@@ -73,15 +71,6 @@ export function getAssociatedGroupsForReference(state: GlobalState, teamId: stri
     const channel = getChannel(state, channelId);
     const locale = getCurrentUserLocale(state);
 
-    if (!haveIChannelPermission(state, {
```

Comment:
This permission check is now done before the selector is called in all the components where it is called